inginious/frontend/templates/task.html

```diff
@@ -222,14 +206,11 @@ <h3>{{ _("Submission history") }}</h3>
                             {# The badge is green (success style) if all tags are validated, blue (info style) otherwise. #}
                             {% if visible_tags|length > 0 %}
                                 {% set ns = namespace(tags_ok_counter=0,list_tags_ok=[]) %}
-                                {% for category in task_categories %}
-                                    {% set tag = course_tags[category] %}
-                                    {% if tag.get_type() == 0 and (tag.is_visible_for_student() or user_manager.has_staff_rights_on_course(course)) %}
-                                        {% set ns.class_tag = "badge alert-info" %}
-                                        {% if "tests" in submission and tag.get_id() in submission["tests"] and submission["tests"][tag.get_id()] %}
-                                            {% set ns.tags_ok_counter = ns.tags_ok_counter + 1 %}
-                                            {% set ns.list_tags_ok = ns.list_tags_ok + tag.get_name(user_manager.session_language()) %}
-                                        {% endif %}
+                                {% for tag in course.get_task_dispenser().get_categories(task.get_id()) %}
```

Comment:
Here we talk about tags and not categories. You can probably keep the old code and  iterate on course_tags.